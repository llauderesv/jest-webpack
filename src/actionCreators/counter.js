//  Actions...
export const INC_COUNTER = 'INC_COUNTER';
export const INC_COUNTER_ASYNC = 'INC_COUNTER_ASYNC';
export const DEC_COUNTER = 'DEC_COUNTER';

// Action creators...
export function incrementCounter() {
  return {
    type: INC_COUNTER,
  };
}

export function decrementCounter() {
  return {
    type: DEC_COUNTER,
  };
}

export function incrementAsyncCounter() {
  return { type: INC_COUNTER_ASYNC };
}
