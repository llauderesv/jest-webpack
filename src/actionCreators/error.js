// Actions...
export const SHOW_ERROR = 'SHOW_ERROR';
export const HIDE_ERROR = 'HIDE_ERROR';

// Action creators...
export function showError(payload) {
  return {
    type: SHOW_ERROR,
    payload,
  };
}

export function hideError() {
  return { type: HIDE_ERROR };
}
