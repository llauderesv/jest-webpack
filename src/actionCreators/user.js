export const USER_FETCH_REQUESTED = 'USER_FETCH';
export const USER_FETCH_SUCCESS = 'USER_FETCH_SUCCESS';
export const USER_FETCH_ERROR = 'USER_FETCH_ERROR';

export function userFetchRequested() {
  return {
    type: USER_FETCH_REQUESTED,
  };
}

export function userFetchSuccess(payload) {
  return {
    type: USER_FETCH_SUCCESS,
    payload,
  };
}

export function userFetchError() {
  return {
    type: USER_FETCH_ERROR,
  };
}
