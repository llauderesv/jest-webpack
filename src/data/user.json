[{
  "id": 1,
  "first_name": "Robinia",
  "last_name": "Perigoe",
  "email": "rperigoe0@nature.com"
}, {
  "id": 2,
  "first_name": "Mycah",
  "last_name": "Abramovic",
  "email": "mabramovic1@blogspot.com"
}, {
  "id": 3,
  "first_name": "Boris",
  "last_name": "Chucks",
  "email": "bchucks2@jiathis.com"
}, {
  "id": 4,
  "first_name": "Benjamen",
  "last_name": "Fairbourne",
  "email": "bfairbourne3@ft.com"
}, {
  "id": 5,
  "first_name": "Beverley",
  "last_name": "MacEllen",
  "email": "bmacellen4@harvard.edu"
}, {
  "id": 6,
  "first_name": "Dolli",
  "last_name": "Collisson",
  "email": "dcollisson5@chron.com"
}, {
  "id": 7,
  "first_name": "Kristin",
  "last_name": "Dowdam",
  "email": "kdowdam6@archive.org"
}, {
  "id": 8,
  "first_name": "Christophe",
  "last_name": "Sowthcote",
  "email": "csowthcote7@comsenz.com"
}, {
  "id": 9,
  "first_name": "Birgitta",
  "last_name": "Scintsbury",
  "email": "bscintsbury8@ask.com"
}, {
  "id": 10,
  "first_name": "Marlena",
  "last_name": "Fallon",
  "email": "mfallon9@cnn.com"
}, {
  "id": 11,
  "first_name": "Annaliese",
  "last_name": "Fletcher",
  "email": "afletchera@cpanel.net"
}, {
  "id": 12,
  "first_name": "Husein",
  "last_name": "Whall",
  "email": "hwhallb@slashdot.org"
}, {
  "id": 13,
  "first_name": "Leonie",
  "last_name": "Backshall",
  "email": "lbackshallc@mozilla.org"
}, {
  "id": 14,
  "first_name": "Donnamarie",
  "last_name": "Lescop",
  "email": "dlescopd@gravatar.com"
}, {
  "id": 15,
  "first_name": "Madelena",
  "last_name": "Chantrell",
  "email": "mchantrelle@usatoday.com"
}, {
  "id": 16,
  "first_name": "Shelton",
  "last_name": "Catford",
  "email": "scatfordf@apple.com"
}, {
  "id": 17,
  "first_name": "Fay",
  "last_name": "Elsmor",
  "email": "felsmorg@deviantart.com"
}, {
  "id": 18,
  "first_name": "Guilbert",
  "last_name": "Cordelette",
  "email": "gcordeletteh@storify.com"
}, {
  "id": 19,
  "first_name": "Elsworth",
  "last_name": "Ollin",
  "email": "eollini@cnet.com"
}, {
  "id": 20,
  "first_name": "Sandro",
  "last_name": "Sirey",
  "email": "ssireyj@globo.com"
}, {
  "id": 21,
  "first_name": "Renell",
  "last_name": "Schiefersten",
  "email": "rschieferstenk@amazonaws.com"
}, {
  "id": 22,
  "first_name": "Amity",
  "last_name": "Clark",
  "email": "aclarkl@mapquest.com"
}, {
  "id": 23,
  "first_name": "Karlene",
  "last_name": "Chilley",
  "email": "kchilleym@imageshack.us"
}, {
  "id": 24,
  "first_name": "Chantal",
  "last_name": "Middleton",
  "email": "cmiddletonn@163.com"
}, {
  "id": 25,
  "first_name": "Warde",
  "last_name": "Franzel",
  "email": "wfranzelo@ucsd.edu"
}, {
  "id": 26,
  "first_name": "Elihu",
  "last_name": "Cawse",
  "email": "ecawsep@sfgate.com"
}, {
  "id": 27,
  "first_name": "Tobin",
  "last_name": "Arkill",
  "email": "tarkillq@cyberchimps.com"
}, {
  "id": 28,
  "first_name": "Dode",
  "last_name": "Newlin",
  "email": "dnewlinr@usda.gov"
}, {
  "id": 29,
  "first_name": "Nathalia",
  "last_name": "Herkess",
  "email": "nherkesss@flickr.com"
}, {
  "id": 30,
  "first_name": "Stern",
  "last_name": "Arnaldo",
  "email": "sarnaldot@fema.gov"
}, {
  "id": 31,
  "first_name": "Harcourt",
  "last_name": "Slany",
  "email": "hslanyu@cbsnews.com"
}, {
  "id": 32,
  "first_name": "Vale",
  "last_name": "Locks",
  "email": "vlocksv@biblegateway.com"
}, {
  "id": 33,
  "first_name": "Sayres",
  "last_name": "Ouldcott",
  "email": "souldcottw@wikia.com"
}, {
  "id": 34,
  "first_name": "Joscelin",
  "last_name": "Lamball",
  "email": "jlamballx@sohu.com"
}, {
  "id": 35,
  "first_name": "Sigismund",
  "last_name": "Shaxby",
  "email": "sshaxbyy@twitpic.com"
}, {
  "id": 36,
  "first_name": "Eamon",
  "last_name": "Seedhouse",
  "email": "eseedhousez@123-reg.co.uk"
}, {
  "id": 37,
  "first_name": "Tibold",
  "last_name": "Minget",
  "email": "tminget10@gov.uk"
}, {
  "id": 38,
  "first_name": "Starlin",
  "last_name": "Bennis",
  "email": "sbennis11@mail.ru"
}, {
  "id": 39,
  "first_name": "Clywd",
  "last_name": "Yakubowicz",
  "email": "cyakubowicz12@nationalgeographic.com"
}, {
  "id": 40,
  "first_name": "Juditha",
  "last_name": "Colleck",
  "email": "jcolleck13@google.es"
}, {
  "id": 41,
  "first_name": "Frazer",
  "last_name": "Greest",
  "email": "fgreest14@indiatimes.com"
}, {
  "id": 42,
  "first_name": "Greggory",
  "last_name": "Macer",
  "email": "gmacer15@cloudflare.com"
}, {
  "id": 43,
  "first_name": "Shae",
  "last_name": "Jonson",
  "email": "sjonson16@eepurl.com"
}, {
  "id": 44,
  "first_name": "Erek",
  "last_name": "Foulds",
  "email": "efoulds17@weebly.com"
}, {
  "id": 45,
  "first_name": "Alden",
  "last_name": "Caldecutt",
  "email": "acaldecutt18@xinhuanet.com"
}, {
  "id": 46,
  "first_name": "Barbara",
  "last_name": "d' Elboux",
  "email": "bdelboux19@reuters.com"
}, {
  "id": 47,
  "first_name": "Leonelle",
  "last_name": "Keigher",
  "email": "lkeigher1a@seattletimes.com"
}, {
  "id": 48,
  "first_name": "Bren",
  "last_name": "McFetridge",
  "email": "bmcfetridge1b@diigo.com"
}, {
  "id": 49,
  "first_name": "Nanete",
  "last_name": "Loy",
  "email": "nloy1c@aol.com"
}, {
  "id": 50,
  "first_name": "Ede",
  "last_name": "Tremeer",
  "email": "etremeer1d@booking.com"
}]