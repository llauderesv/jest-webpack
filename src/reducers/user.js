import {
  USER_FETCH_SUCCESS,
  USER_FETCH_ERROR,
  USER_FETCH_REQUESTED,
} from '../actionCreators/user';

const initialState = { data: [], isFetching: false };

const reducer = (state = initialState, action) => {
  if (action.type === USER_FETCH_REQUESTED) {
    return { ...state, isFetching: true };
  }

  if (action.type === USER_FETCH_SUCCESS) {
    return { ...state, data: action.payload.data, isFetching: false };
  }

  if (action.type === USER_FETCH_ERROR) {
    return { ...state, isFetching: false };
  }

  return state;
};

export default reducer;
