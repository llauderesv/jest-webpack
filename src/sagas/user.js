import axios from 'axios';
import { call, takeEvery, put } from 'redux-saga/effects';
import {
  USER_FETCH_REQUESTED,
  userFetchSuccess,
  userFetchError,
} from '../actionCreators/user';

import userData from '../data/user.json';

export const delay = seconds => new Promise(res => setTimeout(res, seconds));

export function* fetchUser() {
  try {
    // const data = call(axios.get, '')
    const data = { data: userData };
    yield call(delay, 5000);
    yield put(userFetchSuccess(data));
  } catch (error) {
    yield put(userFetchError());
  }
}

export default function* watchFetchUser() {
  yield takeEvery(USER_FETCH_REQUESTED, fetchUser);
}
