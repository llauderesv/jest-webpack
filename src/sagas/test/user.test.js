import 'babel-polyfill';
import { put, call } from 'redux-saga/effects';
import { fetchUser, delay } from '../user';
import { userFetchSuccess } from '../../actionCreators/user';
import userData from '../../data/user.json';

describe('User Saga Tests', () => {
  it('fetchUser Saga test', () => {
    // This will return a plain objects
    // {done: false, value: (The yielded value)}
    const gen = fetchUser();

    // Gen next value will return the yielded function/value
    expect(gen.next().value).toEqual(call(delay, 5000));

    expect(gen.next().value).toEqual(put(userFetchSuccess({ data: userData })));
    // Increment saga must be done...
    expect(gen.next().value).toBeUndefined();
  });
});
