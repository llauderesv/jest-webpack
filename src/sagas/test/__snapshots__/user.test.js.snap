// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Saga Tests fetchUser Saga test 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "payload": Object {
        "data": Array [
          Object {
            "email": "rperigoe0@nature.com",
            "first_name": "Robinia",
            "id": 1,
            "last_name": "Perigoe",
          },
          Object {
            "email": "mabramovic1@blogspot.com",
            "first_name": "Mycah",
            "id": 2,
            "last_name": "Abramovic",
          },
          Object {
            "email": "bchucks2@jiathis.com",
            "first_name": "Boris",
            "id": 3,
            "last_name": "Chucks",
          },
          Object {
            "email": "bfairbourne3@ft.com",
            "first_name": "Benjamen",
            "id": 4,
            "last_name": "Fairbourne",
          },
          Object {
            "email": "bmacellen4@harvard.edu",
            "first_name": "Beverley",
            "id": 5,
            "last_name": "MacEllen",
          },
          Object {
            "email": "dcollisson5@chron.com",
            "first_name": "Dolli",
            "id": 6,
            "last_name": "Collisson",
          },
          Object {
            "email": "kdowdam6@archive.org",
            "first_name": "Kristin",
            "id": 7,
            "last_name": "Dowdam",
          },
          Object {
            "email": "csowthcote7@comsenz.com",
            "first_name": "Christophe",
            "id": 8,
            "last_name": "Sowthcote",
          },
          Object {
            "email": "bscintsbury8@ask.com",
            "first_name": "Birgitta",
            "id": 9,
            "last_name": "Scintsbury",
          },
          Object {
            "email": "mfallon9@cnn.com",
            "first_name": "Marlena",
            "id": 10,
            "last_name": "Fallon",
          },
          Object {
            "email": "afletchera@cpanel.net",
            "first_name": "Annaliese",
            "id": 11,
            "last_name": "Fletcher",
          },
          Object {
            "email": "hwhallb@slashdot.org",
            "first_name": "Husein",
            "id": 12,
            "last_name": "Whall",
          },
          Object {
            "email": "lbackshallc@mozilla.org",
            "first_name": "Leonie",
            "id": 13,
            "last_name": "Backshall",
          },
          Object {
            "email": "dlescopd@gravatar.com",
            "first_name": "Donnamarie",
            "id": 14,
            "last_name": "Lescop",
          },
          Object {
            "email": "mchantrelle@usatoday.com",
            "first_name": "Madelena",
            "id": 15,
            "last_name": "Chantrell",
          },
          Object {
            "email": "scatfordf@apple.com",
            "first_name": "Shelton",
            "id": 16,
            "last_name": "Catford",
          },
          Object {
            "email": "felsmorg@deviantart.com",
            "first_name": "Fay",
            "id": 17,
            "last_name": "Elsmor",
          },
          Object {
            "email": "gcordeletteh@storify.com",
            "first_name": "Guilbert",
            "id": 18,
            "last_name": "Cordelette",
          },
          Object {
            "email": "eollini@cnet.com",
            "first_name": "Elsworth",
            "id": 19,
            "last_name": "Ollin",
          },
          Object {
            "email": "ssireyj@globo.com",
            "first_name": "Sandro",
            "id": 20,
            "last_name": "Sirey",
          },
          Object {
            "email": "rschieferstenk@amazonaws.com",
            "first_name": "Renell",
            "id": 21,
            "last_name": "Schiefersten",
          },
          Object {
            "email": "aclarkl@mapquest.com",
            "first_name": "Amity",
            "id": 22,
            "last_name": "Clark",
          },
          Object {
            "email": "kchilleym@imageshack.us",
            "first_name": "Karlene",
            "id": 23,
            "last_name": "Chilley",
          },
          Object {
            "email": "cmiddletonn@163.com",
            "first_name": "Chantal",
            "id": 24,
            "last_name": "Middleton",
          },
          Object {
            "email": "wfranzelo@ucsd.edu",
            "first_name": "Warde",
            "id": 25,
            "last_name": "Franzel",
          },
          Object {
            "email": "ecawsep@sfgate.com",
            "first_name": "Elihu",
            "id": 26,
            "last_name": "Cawse",
          },
          Object {
            "email": "tarkillq@cyberchimps.com",
            "first_name": "Tobin",
            "id": 27,
            "last_name": "Arkill",
          },
          Object {
            "email": "dnewlinr@usda.gov",
            "first_name": "Dode",
            "id": 28,
            "last_name": "Newlin",
          },
          Object {
            "email": "nherkesss@flickr.com",
            "first_name": "Nathalia",
            "id": 29,
            "last_name": "Herkess",
          },
          Object {
            "email": "sarnaldot@fema.gov",
            "first_name": "Stern",
            "id": 30,
            "last_name": "Arnaldo",
          },
          Object {
            "email": "hslanyu@cbsnews.com",
            "first_name": "Harcourt",
            "id": 31,
            "last_name": "Slany",
          },
          Object {
            "email": "vlocksv@biblegateway.com",
            "first_name": "Vale",
            "id": 32,
            "last_name": "Locks",
          },
          Object {
            "email": "souldcottw@wikia.com",
            "first_name": "Sayres",
            "id": 33,
            "last_name": "Ouldcott",
          },
          Object {
            "email": "jlamballx@sohu.com",
            "first_name": "Joscelin",
            "id": 34,
            "last_name": "Lamball",
          },
          Object {
            "email": "sshaxbyy@twitpic.com",
            "first_name": "Sigismund",
            "id": 35,
            "last_name": "Shaxby",
          },
          Object {
            "email": "eseedhousez@123-reg.co.uk",
            "first_name": "Eamon",
            "id": 36,
            "last_name": "Seedhouse",
          },
          Object {
            "email": "tminget10@gov.uk",
            "first_name": "Tibold",
            "id": 37,
            "last_name": "Minget",
          },
          Object {
            "email": "sbennis11@mail.ru",
            "first_name": "Starlin",
            "id": 38,
            "last_name": "Bennis",
          },
          Object {
            "email": "cyakubowicz12@nationalgeographic.com",
            "first_name": "Clywd",
            "id": 39,
            "last_name": "Yakubowicz",
          },
          Object {
            "email": "jcolleck13@google.es",
            "first_name": "Juditha",
            "id": 40,
            "last_name": "Colleck",
          },
          Object {
            "email": "fgreest14@indiatimes.com",
            "first_name": "Frazer",
            "id": 41,
            "last_name": "Greest",
          },
          Object {
            "email": "gmacer15@cloudflare.com",
            "first_name": "Greggory",
            "id": 42,
            "last_name": "Macer",
          },
          Object {
            "email": "sjonson16@eepurl.com",
            "first_name": "Shae",
            "id": 43,
            "last_name": "Jonson",
          },
          Object {
            "email": "efoulds17@weebly.com",
            "first_name": "Erek",
            "id": 44,
            "last_name": "Foulds",
          },
          Object {
            "email": "acaldecutt18@xinhuanet.com",
            "first_name": "Alden",
            "id": 45,
            "last_name": "Caldecutt",
          },
          Object {
            "email": "bdelboux19@reuters.com",
            "first_name": "Barbara",
            "id": 46,
            "last_name": "d' Elboux",
          },
          Object {
            "email": "lkeigher1a@seattletimes.com",
            "first_name": "Leonelle",
            "id": 47,
            "last_name": "Keigher",
          },
          Object {
            "email": "bmcfetridge1b@diigo.com",
            "first_name": "Bren",
            "id": 48,
            "last_name": "McFetridge",
          },
          Object {
            "email": "nloy1c@aol.com",
            "first_name": "Nanete",
            "id": 49,
            "last_name": "Loy",
          },
          Object {
            "email": "etremeer1d@booking.com",
            "first_name": "Ede",
            "id": 50,
            "last_name": "Tremeer",
          },
        ],
      },
      "type": "USER_FETCH_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`User Saga Tests fetchUser Saga test 2`] = `undefined`;

exports[`User Saga Tests fetchUser Saga test 3`] = `undefined`;
